### PR TITLE
- 100% code coverage - fix some methods - 100% inspection test

### PR DESCRIPTION
**Pull request by MrEssex**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,11 +9,19 @@
     }
   ],
   "require": {
-    "php": ">=7.3"
+    "php": ">=7.4"
```
Comment (bajb) [inline composer.json]: needs to go back to 7.3 as still actively supported

File: composer.lock [hunk 2]
```diff
@@ -0,0 +1,2107 @@
+{
```
Comment (bajb) [inline composer.lock]: Please do not commit your composer.lock

File: tests/ArrayDataMapTest.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+use Packaged\Map\ArrayDataMap;
+use Packaged\Map\DataMap;
+use PHPUnit\Framework\TestCase;
+
+class ArrayDataMapTest extends TestCase
+{
+
+  protected function _simpleMap(): DataMap
+  {
+    $data = new DataMap();
```
Comment (bajb) [inline tests/ArrayDataMapTest.php]: we should not be using DataMap in an ArrayDataMap test

File: tests/ArrayDataMapTest.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    self::assertEquals($array->get('d'), ['a' => 'y', 'z']);
+  }
+
+  public function testAppend(): void
+  {
+    $data = ['a' => '1', 'b' => '2', 'c' => 3];
+    $arrayDataMap = new ArrayDataMap();
+
+    $arrayDataMap->set('1', $data);
+    $equals = [1 => $data];
```
Comment (bajb) [inline tests/ArrayDataMapTest.php:33]: not a big fan of this equals array, esp when it gets to $equals[2].

File: tests/DataMapTest.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+    self::assertEquals(3, $count);
+    self::assertEmpty($comparison);
+  }
+
+  public function testCount(): void
+  {
+    $data = $this->_simpleMap();
+    self::assertEquals(3, $data->count());
+    self::assertNotEquals(4, $data->count());
```
Comment (bajb) [inline tests/DataMapTest.php]: this is already guaranteed by the previous assertion

File: tests/DataMapTest.php [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    $data = $this->_simpleMap();
+    self::assertEquals(['fruit' => 'apple', 'color' => 'red', 'dog' => 'poodle'], $data->all());
+    self::assertNotContains(['color' => 'red', 'fruit' => 'apple', 'dog' => 'poodle'], $data->all());
+  }
+
+  public function testClear(): void
+  {
+    $data = $this->_simpleMap();
+    self::assertEquals(['fruit' => 'apple', 'color' => 'red', 'dog' => 'poodle'], $data->all());
+    self::assertEquals([], $data->clear()->all());
```
Comment (bajb) [inline tests/DataMapTest.php:138]: test count as well

File: tests/DataMapTest.php [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    self::assertTrue($data->has('color'));
+    self::assertTrue($data->has('dog'));
+    $data->remove('dog');
+    self::assertFalse($data->has('dog'));
+  }
+
+  public function testGetInt(): void
+  {
+    $data = $this->_simpleMap();
+    $data->set(1, 2);
```
Comment (bajb) [inline tests/DataMapTest.php]: Maybe use more than 1 and 2 here.   A default value of 5 or something would avoid incorrect results

File: tests/ArrayDataMapTest.php (hunk 4 above)
Comment (MrEssex) [reply tests/ArrayDataMapTest.php:33]: Can you provide an alternative? :)

**Pull request merged**